mysql-test/suite/galera/t/galera_sst_xtrabackup-v2_super_read_only.test

```diff
@@ -1,26 +1,97 @@
 #
-# This test checks the pxc-encrypt-cluster-traffic option (auto SSL config).
-# Initial SST happens via xtrabackup, so there is not much to do in the body of the test
+# This test checks that the cluster starts up (without error)
+# if super-read-only=ON.  This will check server startup
+# as well as the SST process.
 #
 
 --source include/big_test.inc
 --source include/galera_cluster.inc
 
+--let $wait_condition = SELECT VARIABLE_VALUE = 2 FROM performance_schema.global_status WHERE VARIABLE_NAME = 'wsrep_cluster_size';
+--source include/wait_condition.inc
+
 SELECT 1;
 
+# --------------------------------
+# Test 1: Test that the cluster starts up
+# --------------------------------
+
+--connection node_2
+--echo [connection node_2]
+
+--let $NODE2_CONFIG_FILE = $MYSQLTEST_VARDIR/tmp/galera_sst_xtrabackup-v2_super_read_only.test.cnf
+--copy_file $MYSQLTEST_VARDIR/my.cnf $NODE2_CONFIG_FILE
+--append_file $NODE2_CONFIG_FILE
+	[mysqld]
+	super_read_only=ON
+EOF
+
+# --------------------------------
+# Test 1a: Restart node2 (no sst)
+# --------------------------------
+
+# restart the server with super-read-only=ON (no SST)
+--echo "# restarting node2 (no SST)"
+--let $restart_hide_args=1
+--let $restart_parameters = "restart: --defaults-file=$NODE2_CONFIG_FILE"
+--source include/restart_mysqld.inc
+--source include/wait_until_connected_again.inc
+
+SET SESSION wsrep_sync_wait = 0;
 --let $wait_condition = SELECT VARIABLE_VALUE = 2 FROM performance_schema.global_status WHERE VARIABLE_NAME = 'wsrep_cluster_size';
 --source include/wait_condition.inc
 
---connection node_2
+SELECT 1;
+
+# We should not find the super-read-only warnings in the logs
+--let $assert_text = Could not find warnings about super-read-only
+--let $assert_select = super-read-only
+--let $assert_count = 0
+--let $assert_file = $MYSQLTEST_VARDIR/log/mysqld.2.err
+--let $assert_only_after = CURRENT_TEST
+--source include/assert_grep.inc
+
+
+# --------------------------------
+# Test 1b: Restart node2 (with sst)
+# --------------------------------
+--echo "# shutting down node2"
 SET SESSION wsrep_sync_wait = 0;
 --source include/shutdown_mysqld.inc
+--source include/wait_until_disconnected.inc
+
+# remove the grastate.dat file to force an SST
 --remove_file $MYSQLTEST_VARDIR/mysqld.2/data/grastate.dat
---exec sed -i '/^\[mysqld.2\]/a super_read_only=ON' $MYSQLTEST_VARDIR/my.cnf
+
+# restart the server with super-read-only=ON (SST)
+--echo "# restarting node2 (with SST)"
+--let $restart_hide_args=1
+--let $start_mysqld_params = "--defaults-file=$NODE2_CONFIG_FILE"
 --let $_expect_file_name= $MYSQLTEST_VARDIR/tmp/mysqld.2.expect
 --source include/start_mysqld.inc
+--source include/wait_until_connected_again.inc
```

Comment:
Same as above.
```
+ # restart the server with super-read-only=ON (SST)
+ --echo "# restarting node2 (with SST)"
+ --let $restart_parameters = "restart: --super-read-only=ON"
  --let $_expect_file_name= $MYSQLTEST_VARDIR/tmp/mysqld.2.expect
  --source include/start_mysqld.inc
```

Reply:
The SST script does not have access to all of the command-line parameters, so it reads from the config file.  Thus we need the config file to test how the SST handles the super-read-only option.